### ISSUE TITLE
Add accessibility to form components

### ISSUE THREAD

**Issue by aleavikraman** [epicmaxco/vuestic-ui#3104]
**Is your feature request related to a problem? Please describe.**

When trying to add accessibility such as aria labels to form fields it's not always built in a way that allows for easy placement.  For example trying to add an aria-label to va-select it does not get read out to a screen reader

**Describe the solution you'd like**

ability to update focus, add aria labels and any other accessibility should be included in form components. 

**Describe alternatives you've considered**

**Additional context**

Let me know if there are plans to implement accessibility into components, I'd like to try and help with that.

**Comment by aleavikraman** [comment]
That would probably work, but it might be overkill unless it's a complex component or there's a lot of accessibility options that can be passed in.   I guess I was just thinking more of being able to pass in an aria label and aria labeledby and having access to the focus variable. 

So for `<va-select aria-label="choose subscription">` would read out as "choose subscription" when it's tabbed on instead of "button" which is what I think it is now. 

For example on the papa johns website when you tab over this icon it reads "choose your language" instead of reading that its just a dropdown or button. I want to be able to implement an aria label so I can achieve the same thing. 

<img width="1212" alt="Screenshot 2023-03-02 at 11 11 38 AM" src="https://user-images.githubusercontent.com/16092098/222485853-c0e3c793-eda1-4a88-929e-795a20f83645.png">

let me know if you need help making a todo list, but just having an aria-label/labeledby option on most components and being able to have access to change the focus variable color would be a huge start. 

On input error if I could change the color of the text separate from the border color that would be great too since the color ratio that passes a11y compliance is different between the border and text. 

For example the red border around an input to the background will pass with a color contrast ration of 3:1 
but the text color needs a ratio of 4:5:1

**Comment by m0ksem** [comment]
@aluarius, do you have any thought on it? I feel like having just aria-label is not enough, at least for VaSelect, because we have at least three interactive elements: select input, reset button, filter input. The same goes to VaDatePicker.

We might just have a lot of props with aria prefix...
Here are few examples:
```vue
<va-select aria-label="Choose subscription" aria-reset-label="Clear choosen subscription" />
```
For this two this can be done with slots actually...
```vue
<va-modal aria-ok-label="Confirm subscription" aria-cancel-label="Close confirmation window" />
```

```vue
<va-counter aria-minus-button="Remove cup of coffee" aria-plus-button="Add cup of coffee"  />
```

**Comment by m0ksem** [comment]
> On input error if I could change the color of the text separate from the border color that would be great too since the color ratio that passes a11y compliance is different between the border and text.

You should be able to do this with CSS. We use Bem, so you can do 
```css
  .va-input--error { color: #f0f; border-color: #0f0; }
```

**Comment by aleavikraman** [comment]
Oh that's interesting I didn't realize the feature of typing to make a selection was there. I'm not sure a screen reader or keyboard only user would know that feature exists out of the box. When I used it with a screen reader it didn't read out the word to me so how would I know it's selected (assuming I can't see the selection being made)? 

I'm not sure what the reset functionality is. How would a screen reader or keyboard only user use that? Normally if I wanted to make a new selection I would just tab back into the select and choose something else

For the input error I did try to override the styles and I also tried using the code you provided, but it looks like the color might not properly be attached to that? I think its part of `.va-message-list` but I wasn't able to override it

<img width="639" alt="Screenshot 2023-03-02 at 5 34 45 PM" src="https://user-images.githubusercontent.com/16092098/222575400-fc5ff679-d06e-458e-9eb3-a4754b510ce7.png">


I'm using     "@vuestic/nuxt": "^1.0.11",  in case that makes a difference 

Btw thanks for helping me out with this!

**Comment by aluarius** [comment]
See also https://github.com/epicmaxco/vuestic-ui/issues/2863

**Comment by aluarius** [comment]
Have decided to create a composable, that will allow to set aria-label, aria-labelledby, aria-describedby and aria-hidden attributes (mb something else, research is needed) to any Vuestic UI component.

**Comment by m0ksem** [comment]
Hey, @aleavikraman. I just made a PR to resolve this issue. LMK if it solves your problem:
https://github.com/epicmaxco/vuestic-ui/pull/3175
